.github/workflows/linuxtests.yml

```diff
@@ -0,0 +1,50 @@
+name: Linux Tests
+
+on:
+  push:
+    branches:
+      - master
+      - maintenance/**
+  pull_request:
+    branches:
+      - master
+      - maintenance/**
+
+jobs:
+  Python-37-dbg:
+    name: Python 3.7-dbg
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Configure env vars and python packages
+        run: |
+          sudo apt install python3.7-dbg python3.7-dev libatlas-base-dev liblapack-dev gfortran libgmp-dev libmpfr-dev libsuitesparse-dev ccache swig libmpc-dev
+          free -m
+          export PATH=/usr/lib/ccache:$PATH
+          ccache --zero-stats
+          export CCACHE_COMPRESS=1
+          python3.7-dbg --version # just to check
+          export NPY_NUM_BUILD_JOBS=2
+          uname -a
+          df -h
+          ulimit -a
+          set -e -o pipefail
+          mkdir builds
+          cd builds
+          python3.7-dbg -c 'import sys; print("Python debug build:", hasattr(sys, "gettotalrefcount"))'
+          python3.7-dbg -m pip install --upgrade pip setuptools wheel cython pytest pytest-xdist argparse pybind11
+          python3.7-dbg -m pip install --upgrade mpmath gmpy2
+          python3.7-dbg -m pip install --upgrade numpy
+          python3.7-dbg -m pip uninstall -y nose
+          ccache -s
```

Comment:
I don't think ccache is going to do anything to speed up the scipy build because the cache isn't being uploaded as an artefact, or such an artefact being restored, from this script.

Reply:
I actually don't know much about ccache, I was blindly copying. I can remove if you think this won't make any difference.

Reply:
`ccache` can be made to work: https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/

could be left as a follow-up though.

Reply:
I've used ccache+gh on a different project, and then tried to have a go here. I couldn't get it to work out of the box, probably because I don't know enough about the build process.
It can be left for a follow up, but I'd prefer for the lines to be removed because otherwise we think it's doing something.

Reply:
I left it for a follow-up and for a more knowledgable person.